text/3712-box-yesalias.md

```diff
@@ -0,0 +1,115 @@
+
+In the past, the operational semantics team has considered many ad-hoc solutions to the problem, while maintaining special cases in the aliasing model (such as Weak Protectors) that only exist for `Box<T>`. 
+For example, moving a `ManuallyDrop<Box<T>>` after calling `Drop` is immediate undefined behaviour (due to the `Box` no longer being dereferenceable) - <https://rust-lang.zulipchat.com/#narrow/stream/136281-t-opsem/topic/Moving.20.60ManuallyDrop.3CBox.3C_.3E.3E.60>, and the Active tag requirements for a `Box<T>` are unsound when combined with custom allocators <https://rust-lang.zulipchat.com/#narrow/stream/213817-t-lang/topic/Is.20Box.20a.20language.20type.20or.20a.20library.20type.3F>. This wastes procedural time reviewing the proposals, and complicates the language by introducing special case rules that would otherwise be unnecessary.
+
+Any `unsafe` code that may want to temporarily maintain aliased `Box<T>`s for various reasons (such as low-level copy operations), or may want to use something like `ManuallyDrop<Box<T>>`, is put into an interesting position: While they can use a user-defined smart pointer, this requires both care on the part of the smart pointer implementor, but also affects the ergonomics and expressiveness of that code, as `Box<T>` has many special language features that surface at the syntax level, which cannot be replicated today by a user-defined type.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+[`alloc::boxed::Box<T>`] is defined with a number of magic features that interact directly with the language. To limit the impact, both on users and on the language itself, we remove the language level requirement that `Box` be dereferenceable and unique (like applies to `&mut T`). 
+This does not affect library level requirements of uniqueness however - it may remain undefined behaviour to access a `Box` that is being aliased (particularily where the operations produce multiple aliasing mutable references from different `Box`es) or to use `Box::from_raw` to construct multiple aliasing boxes. 
+
+We also do not remove any other language level magic from `Box`, such as the ability to do both partial and complete moves from a `Box`.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+For the remainder of this section, let `WellFormed<T>` designate a type for *exposition purposes only* defined as follows:
+```rust
+#[repr(transparent)]
+struct WellFormed<T: ?Sized>(core::ptr::NonNull<T>);
+```
+
+(Note that we do not define this type in the public standard library interface, though an implementation of the standard library could define the type locally)
+
+The following are not valid values of type `WellFormed<T>`, and a typed copy that would produce such a value is undefined behaviour:
```

Comment:
The Reference has been adjusted a while ago to state validity invariants positively, i..e by listing what must be true, instead of listing what must not be false. IMO that's more understandable, and the RFC should be updated to also do that.